src/Symfony/Component/Routing/Matcher/Dumper/StaticPrefixCollection.php

```diff
@@ -118,41 +118,41 @@ public function populateCollection(RouteCollection $routes): RouteCollection
     /**
      * Tries to combine a route with another route or group.
      */
-    private function groupWithItem(int $i, string $prefix, $route): ?self
+    private function groupWithItem(int $i, string $prefix, string $staticPrefix, $route): bool
     {
-        if (!$commonPrefix = $this->detectCommonPrefix($prefix, $this->prefixes[$i])) {
-            return null;
+        list($commonPrefix, $commonStaticPrefix) = $this->getCommonPrefix($prefix, $this->prefixes[$i]);
+
+        if (strlen($this->prefix) >= strlen($commonPrefix)) {
```

Comment:
Not sure how often this is called or if performance matters, but the prefix property doesn't change from what I can tell, should the result be cached? There are a few there are a few more uses of this property like this.

Reply:
fixed by adding a `\`, nothing else to gain from caching the length